eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -2292,4 +2293,14 @@ default void appendString(Appendable appendable, String separator)
      * @since 3.0
      */
     <K, V> MapIterable<K, V> aggregateBy(Function<? super T, ? extends K> groupBy, Function0<? extends V> zeroValueFactory, Function2<? super V, ? super T, ? extends V> nonMutatingAggregator);
+
+    /**
+     * Dynamically alters the short-circuiting behavior on anySatisfy and detect
+     *
+     * @since 10.0.
+     */
+    default RichIterable<T> asNonShortCircuit()
```

Comment:
This is an interesting part. Other wrappers (synchronized, unmodifiable) have subclasses for every sub-interface. But short circuit methods never return collection types, so it doesn't feel worth in. Now I'm not sure if this should return RichIterable either. It could return a smaller interface with only the methods that normally short circuit.

Reply:
There's a new interface ShortCIrcuitIterable